WordPress/gutenberg
# Components: Refactor `ColorPalette` tests to `@testing-library/react`

## What?
We've recently started refactoring `enzyme` tests to `@testing-library/react`.

This PR refactors the `ColorPalette` tests from `enzyme` to `@testing-library/react`.

## Why?
`@testing-library/react` provides a better way to write tests for accessible components that is closer to the way the user experiences them.

## How?
We're straightforwardly replacing `enzyme` tests with `@testing-library/react` ones, using `jest-dom` matchers and mocks to avoid testing unrelated implementation details.

Ideally, we should not be using snapshot tests here, however, the purpose of this PR is not to improve or enrich the tests themselves, but rather to migrate them `@testing-library/react`. Our primary motivation is unblocking the upgrade to React 18. 

In any case, it would be a good future step to improve the tests and test actual experience and interaction in order to improve the overall value of the tests.

## Testing Instructions
Verify tests pass: `npm run test:unit packages/components/src/color-palette/test/index.js`
